MergerCli/Process.cs

```diff
@@ -4,57 +4,82 @@
 using MergerLogic.ImageProcessing;
 using MergerLogic.Utils;
 using Microsoft.Extensions.Logging;
+using System.Collections.Concurrent;
 
 namespace MergerCli
 {
     internal class Process : IProcess
     {
         private Func<Coord, Tile?> _getTileByCoord;
-
-        private readonly IConfigurationManager _config;
+        private readonly IConfigurationManager _configManager;
         private readonly ITileMerger _tileMerger;
         private readonly ILogger _logger;
 
         public Process(IConfigurationManager configuration, ITileMerger tileMerger, ILogger<Process> logger)
         {
-            this._config = configuration;
+            this._configManager = configuration;
             this._tileMerger = tileMerger;
             this._logger = logger;
         }
 
         public void Start(TileFormat targetFormat, IData baseData, IData newData, int batchSize,
             BatchStatusManager batchStatusManager)
         {
-            batchStatusManager.InitializeLayer(newData.Path);
-            List<Tile> tiles = new List<Tile>(batchSize);
+            // ConcurrentBag<Tile> tiles = new ConcurrentBag<Tile>();
             long totalTileCount = newData.TileCount();
+            batchStatusManager.InitializeLayer(newData.Path);
             long tileProgressCount = 0;
+            bool resumeMode = false;
+            bool pollForBatch = true;
```

Comment:
what is the porpuss of the 'pollForBatch' variable

Reply:
its a flag for ParallelRun, used in while statement  (tileProgressCount != totalTileCount && pollForBatch) to prevent threads spam for polling tasks if tiles count is 0